src/server/editorServices.ts

```diff
@@ -3163,32 +3208,37 @@ namespace ts.server {
             // Work on array copy as we could add more projects as part of callback
             for (const project of arrayFrom(this.configuredProjects.values())) {
                 // If this project has potential project reference for any of the project we are loading ancestor tree for
-                // we need to load this project tree
-                if (forEachPotentialProjectReference(
-                    project,
-                    potentialRefPath => forProjects!.has(potentialRefPath)
-                ) || forEachResolvedProjectReference(
-                    project,
-                    (_ref, resolvedPath) => forProjects!.has(resolvedPath)
-                )) {
-                    // Load children
-                    this.ensureProjectChildren(project, seenProjects);
+                // load this project first
+                if (forEachPotentialProjectReference(project, potentialRefPath => forProjects!.has(potentialRefPath))) {
+                    updateProjectIfDirty(project);
                 }
+                this.ensureProjectChildren(project, forProjects, seenProjects);
             }
         }
 
-        private ensureProjectChildren(project: ConfiguredProject, seenProjects: Set<NormalizedPath>) {
+        private ensureProjectChildren(project: ConfiguredProject, forProjects: ReadonlyCollection<string>, seenProjects: Set<NormalizedPath>) {
             if (!tryAddToSet(seenProjects, project.canonicalConfigFilePath)) return;
-            // Update the project
-            updateProjectIfDirty(project);
-
-            // Create tree because project is uptodate we only care of resolved references
-            forEachResolvedProjectReferenceProject(
-                project,
-                child => this.ensureProjectChildren(child, seenProjects),
-                ProjectReferenceProjectLoadKind.FindCreateLoad,
-                `Creating project for reference of project: ${project.projectName}`
-            );
+
+            // If this project disables child load ignore it
+            if (project.getCompilerOptions().disableReferencedProjectLoad) return;

```

Comment:
I had trouble reasoning about which flag to check the flag on.  This is the project in which f12 was pressed?

Reply:
This the project whose children as being explored for loading. Not the project in which definition is found.

Reply:
So, suppose I have projects A, B, and C.  A references both B and C.  I have project B open and I invoke find-all-references on a declaration in B.  Which, if any, flags does the LS look at when deciding whether to load A (which depends on B and may, therefore, have a reference to the declaration)?  Same question for C.

I think the answer is that B's flag determines whether or not A is loaded and C is loaded unconditionally (assuming A is loaded) because A depends on it?

Reply:
A is loaded only if B doesnt have `disableSolutionSearching` is enabled. C is loaded if A is allowed to load and doesnt have `disableReferencedProjectLoad`

Reply:
With this PR C is loaded only if it references B btw and not otherwise

Reply:
I suspect we're stuck with it now, but I find it confusing having these settings apply per-project and not to the whole workspace (e.g. as editor settings).

Reply:
@sheetalkamat I think your responses assumed that A was in a parent directory.  If A, B, and C are in sibling directories, it looks like `disableSolutionSearching` has no effect on whether A is loaded.

Reply:
`disableSolutionSearching` only affects in finding solution in parent directory... (just like we find configured project in parent directory) `disableReferencedProjectLoad` affects loading referenced projects from solution/project the setting is in,